Marlin/src/gcode/feature/pause/M600.cpp

```diff
@@ -43,73 +48,154 @@
+    }      
+  #endif
+	
+  #if ENABLED(PAUSE_HOME_BEFORE)
     // Don't allow filament change without homing first
     if (axis_unhomed_error()) home_all_axes();
   #endif
 
   // Initial retract before move to filament change position
   const float retract = parser.seen('E') ? parser.value_axis_units(E_AXIS) : 0
-    #ifdef PAUSE_PARK_RETRACT_LENGTH
-      - (PAUSE_PARK_RETRACT_LENGTH)
+    #ifdef PAUSE_RET
+      - (PAUSE_RET)
     #endif
   ;
 
   // Lift Z axis
   const float z_lift = parser.linearval('Z', 0
-    #ifdef PAUSE_PARK_Z_ADD
-      + PAUSE_PARK_Z_ADD
+    #ifdef PAUSE_Z_ADD
+      + PAUSE_Z_ADD
     #endif
   );
 
   // Move XY axes to filament exchange position
   const float x_pos = parser.linearval('X', 0
-    #ifdef PAUSE_PARK_X_POS
-      + PAUSE_PARK_X_POS
+    #ifdef PAUSE_X_POS
+      + PAUSE_X_POS
+	#else
+      + current_position[X_AXIS]
     #endif
   );
   const float y_pos = parser.linearval('Y', 0
-    #ifdef PAUSE_PARK_Y_POS
-      + PAUSE_PARK_Y_POS
-    #endif
+    #ifdef PAUSE_Y_POS
+      + PAUSE_Y_POS
+	#else
+      + current_position[Y_AXIS]
+    #endif	
   );
-
+  
   // Unload filament
-  const float unload_length = parser.seen('U') ? parser.value_axis_units(E_AXIS) : 0
-    #if defined(FILAMENT_CHANGE_UNLOAD_LENGTH) && FILAMENT_CHANGE_UNLOAD_LENGTH > 0
-      - (FILAMENT_CHANGE_UNLOAD_LENGTH)
-    #endif
-  ;
-
+  #if ENABLED(PAUSE_SPOOL_SWAP)
+      float unload_length = parser.seen('U') ? parser.value_axis_units(E_AXIS) : 0
+      #ifdef PAUSE_UNLOAD
+        - PAUSE_UNLOAD ;
+      if (swap_spool_enabled) unload_length = -SPOOL_SWAP_UNLOAD;
+      #endif 
+  #else
+      const float unload_length = parser.seen('U') ? parser.value_axis_units(E_AXIS) : 0
+      #ifdef PAUSE_UNLOAD
+        - PAUSE_UNLOAD;    
+      #endif 
+  #endif//end unload filament
+ 
   // Load filament
-  const float load_length = parser.seen('L') ? parser.value_axis_units(E_AXIS) : 0
-    #ifdef FILAMENT_CHANGE_LOAD_LENGTH
-      + FILAMENT_CHANGE_LOAD_LENGTH
-    #endif
-  ;
+  #if ENABLED(PAUSE_SPOOL_SWAP)
+    float load_length = parser.seen('L') ? parser.value_axis_units(E_AXIS) : 0
+    #ifdef PAUSE_LOAD
+	+ PAUSE_LOAD ;
+    if (swap_spool_enabled) load_length = SPOOL_SWAP_LOAD;
+    #endif 
+  #else    
+    const float load_length = parser.seen('L') ? parser.value_axis_units(E_AXIS) : 0
+    #ifdef PAUSE_LOAD
+	  + PAUSE_LOAD ;
+    #endif    
+  #endif //end load filament
 
   const int beep_count = parser.intval('B',
-    #ifdef FILAMENT_CHANGE_NUMBER_OF_ALERT_BEEPS
-      FILAMENT_CHANGE_NUMBER_OF_ALERT_BEEPS
+    #ifdef PAUSE_BEEPS
+      PAUSE_BEEPS
     #else
       -1
     #endif
   );
 
   const bool job_running = print_job_timer.isRunning();
 
-  if (pause_print(retract, z_lift, x_pos, y_pos, unload_length, beep_count, true)) {
-    wait_for_filament_reload(beep_count);
-    resume_print(load_length, ADVANCED_PAUSE_EXTRUDE_LENGTH, beep_count);
-  }
+  #if ENABLED(PAUSE_SPOOL_SWAP)
+    // Next tool 
```

Comment:
Here is the secret of the swap
In M600 , we can call , pause print , and resume print
and between them , i make tool change , flow copy , fwretract copy  and temp copy